src/layers/domain/response/validation_errors.py

```diff
@@ -70,14 +77,60 @@ def decorator(*args: P.args, **kwargs: P.kwargs):
     return decorator
 
 
+class DeviceType(str, Enum):
+    PRODUCT = "PRODUCT"
+    ENDPOINT = "ENDPOINT"
+
+    @classmethod
+    def _missing_(cls, value):
+        for member in cls:
+            if value.upper() == member.value:
+                return member
+
+
+class SearchQueryParams(BaseModel, extra=Extra.forbid):
+    device_type: DeviceType
+
+
+def validate_query_params(function):
+    @wraps(function)
+    def decorator(data, cache, **kwargs):
+        event = APIGatewayProxyEvent(data[StepChain.INIT])
+        query_params = event.query_string_parameters
+        try:
+            SearchQueryParams(**query_params)
+            return function(data, cache, **kwargs)
+        except ValidationError as exc:
+            raise InboundQueryValidationError(
+                errors=exc.raw_errors,
+                model=exc.model,
+            )
+
+    return decorator
```

Comment:
I'd write this code directly into the search step as it's not really a part of the response validation errors utils. 

Also don't think we need a specific `InboundQueryValidationError` - I can see below that you've set the message to `"'device_type' query parameter must be one of 'product' or 'endpoint'` but that's already pretty close to the default message

```
value is not a valid enumeration member; permitted: 'product', 'endpoint' (type=type_error.enum; enum_values=[<DeviceType.PRODUCT: 'product'>, <DeviceType.ENDPOINT: 'endpoint'>])
```